internal/awslambda/extension/manager.go

```diff
@@ -0,0 +1,386 @@
+		)
+	}
+
+	id, exist := resp.Header[api.LambdaExtensionIdentifierHeaderKey]
+	if !exist {
+		return multierr.Combine(
+			errors.New("missing required indentifier header in response"),
+			ErrFailedRegistration,
+		)
+	}
+	// Once we have successfully registered to the lambda,
+	// the id assigned to the process needs to be preserved and sent
+	// with future requests
+	m.registeredID = id[0]
+
+	var info api.RegisterResponsePayload
+	if err := json.NewDecoder(resp.Body).Decode(&info); err != nil {
+		return err
+	}
+
+	// Log the registered payload here as an informative means of
+	// debugging connecitivity issues in future.
+	m.log.WithFields(map[string]interface{}{
+		"function-name":    info.FunctionName,
+		"function-version": info.FunctionVersion,
+		"function-handler": info.Handler,
+	}).Info("Successfully registered with Lambda")
+	return nil
+}
+
+func (m *manager) nextEvent(ctx context.Context) (*api.EventNextPayload, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, api.EventEndpoint.GetUrl(m.domain), http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set(api.LambdaExtensionIdentifierHeaderKey, m.registeredID)
+
+	resp, err := m.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// All things are good in the world
+	default:
+		return nil, multierr.Combine(
+			fmt.Errorf("issue handling request with status code %d", resp.StatusCode),
+			ErrIssueProgress,
+		)
+	}
+
+	var payload api.EventNextPayload
+	if err := json.NewDecoder(resp.Body).Decode(&payload); err != nil {
+		return nil, err
+	}
+
+	return &payload, nil
+}
+
+func (m *manager) initError(ctx context.Context, problem error) error {
+	var (
+		buf bytes.Buffer
+		enc = json.NewEncoder(&buf)
+	)
+
+	enc.SetEscapeHTML(true)
+
+	err := enc.Encode(api.ErrorRequest{
+		Message:    problem.Error(),
+		Type:       "init.failed.additional.invocations",
+		StackTrace: strings.Fields(string(debug.Stack())),
+	})
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, api.InitErrorEndpoint.GetUrl(m.domain), &buf)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set(api.LambdaExtensionIdentifierHeaderKey, m.registeredID)
+	req.Header.Set(api.LambdaErrorHeaderKey, "extension.failed-additional-init")
+
+	resp, err := m.client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if _, err := io.Copy(io.Discard, resp.Body); err != nil {
+		return err
+	}
+	_ = resp.Body.Close()
+
+	if resp.StatusCode == http.StatusAccepted {
+		return nil
+	}
+
+	return fmt.Errorf("issue with sending init error to lambda, status code: %d", resp.StatusCode)
+}
+
+func (m *manager) exitError(ctx context.Context, problem error) error {
+	var (
+		buf bytes.Buffer
+		enc = json.NewEncoder(&buf)
+	)
+
+	enc.SetEscapeHTML(true)
+
+	err := enc.Encode(api.ErrorRequest{
+		Message:    problem.Error(),
+		Type:       "extension.shutdown.failed",
```

Comment:
Nope, it's not standard, it can be any string and is meant to the be an indication that this extension is the source of the failure. I could probably change it to something more in line with their recommendations like `Extension.GostatsdShutdownFailed`